### DISCUSSION TITLE
Assignment - Create a use-case diagram for ReScaler.

### DISCUSSION THREAD

**Discussion by kanmaytacker** [kanmaytacker/design-questions#14] [General]
What?
- Draw a use case diagram for Scaler Academy.
- Your use case diagram should have at least 5 use cases.
- Your use case diagram should have at least 2 actors.
- Introduce a use case that should depend on another use case. (Use include relationship)
- Introduce a use case that should extend another use case. (Use extend relationship)

Why?
Practice UML use case diagram.

How?
* pen and paper
* Simply a list of actors and their usecases should be enough as well

**Comment by kangkanpaul** [comment]
![Untitled](https://user-images.githubusercontent.com/22132740/210409281-63c7e5db-ffb1-43da-b377-5c8777bd6dbe.png)

**Reply by kanmaytacker** [reply]
Hey @kangkanpaul. The use cases look good.
Try to add relations with actors to get a better idea.

Also, the includes relationship seems a little off.
The use case take a class does not necessarily depend on the attend class usecase.

**Comment by ajayv1** [comment]
![IMG_20230103_225339](https://user-images.githubusercontent.com/11134190/210409420-13272e8e-7000-4227-8675-14da5c7dc15e.jpg)

**Reply by kanmaytacker** [reply]
Hey @ajayv1.
These are a good set of use-cases. Try to use verbs though like `Take a class` and `Attend a class`.
Then there would be two different use-cases instead of the one you currently have.

**Comment by sahilshaikh1634** [comment]
![Rescaler](https://user-images.githubusercontent.com/35573387/210410160-7fb468be-451f-4c3d-8acb-219a3448f3ad.jpg)

**Reply by kanmaytacker** [reply]
@sahilshaikh1634 This diagram is almost perfect. Nice!
Few minor things to point out -
* Mention extends and includes on the dashed lines
* While it works, you ideally will not have two different use cases for different types of classes.

**Comment by Vishal70854** [comment]
![1672766653310](https://user-images.githubusercontent.com/48087356/210410642-14f8b572-ae7f-4a54-93d3-2082a309ccf2.jpg)

**Comment by jcr257** [comment]
![6d21e2e3-408c-4b01-a836-172b055b99e1](https://user-images.githubusercontent.com/121890968/210410598-9b235a97-be90-433d-b85b-98300dc2b275.jpg)

**Comment by MRKarthick** [comment]
Usecases:
1.  Login (All can login)
2. Seeing recorded Session (Starting a Recording, Seeing a recording). Though all can see recording only Mentors can start/edit recording
3. Creating, Viewing and Helping for a problem (All can help / viewing while both mentors and Tasks can create problems)
4. Accessing Dashboard. (Only Mentors can access other Students dashboard)
5. Streak (only for students can have their own streak. Mentors and TAs can see students streak)


Actors: Students, Mentors, TechnicalAssistant


Includes Relationship: 
1. Viewing of a problem depends on Creating a problem
2. Viewing recorded session includes creation of session

Extends Relationship: 
1. Accessing student’s dashboard can be done by both student as well as Mentor

**Comment by kishor-a** [comment]
** Actors **

Student
Instructor
Mentor
Admin
SalesPerson
TA
PlacementCoordinator

** Use cases **

Student:
	login, attendClass, solveAssignment, solveHomework, raiseHelpRequest, updateProfile

Instructor:
	login, takeClass, solveDoubts, createAssignment, 

Admin:
	login, scheduleClass, rescheduleClass, solveTicket, onboardStudent

TA:
	solveHelpRequest, guideStudent

** Includes relations **

solveHelpRequest - - - -includes- - - - raiseHelpRequest

solveAssignment  - - - - includes - - - - createAssignment

takeClass - - - - - includes - - - - scheduleClass

**Comment by AdityaMhetras** [comment]
![WhatsApp Image 2023-01-03 at 11 09 15 PM](https://user-images.githubusercontent.com/119006079/210411476-2965ca6f-9a10-4569-a86c-c58b60bb700e.jpeg)

**Comment by dhruvilad** [comment]
![IMG-3849](https://user-images.githubusercontent.com/47177026/210411676-9bc889cc-3711-4581-bec3-d7026fabf1b5.jpg)

**Comment by rafiqul0396** [comment]
![ScalerDiagrman](https://user-images.githubusercontent.com/76397310/210412539-7c90d4a6-aa52-4d53-9ad9-87f6aaf1f244.png)
